django/db/models/functions.py

```diff
@@ -83,6 +83,28 @@ def _paired(self, expressions):
         return ConcatPair(expressions[0], self._paired(expressions[1:]))
 
 
+class Greatest(Func):
+    """
+    Chooses the maximum non-null expression and returns it.
+    """
+    function = 'GREATEST'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) < 1:
+            raise ValueError('Greatest must take at least one expression')
+        super(Greatest, self).__init__(*expressions, **extra)
+
+    def as_sqlite(self, compiler, connection):
+        self.coalesce()
+        return super(Greatest, self).as_sql(compiler, connection, function='MAX')
+
+    def coalesce(self):
+        # any null results in null for the whole expression, wrap with coalesce
+        expressions = [
+            Coalesce(expression, Value('')) for expression in self.get_source_expressions()]
+        self.set_source_expressions(expressions)
```

Comment:
This has been copied from `ConcatPair` above. I'm going to need it for `Least` as well. To avoid repetition, I'd like to put it somewhere unique, but I'm not sure whether a `Mixin` or a custom subclass of `Func` or something else is most appropriate.